### PR TITLE
Fix #14676: Athena S3 Lineage

### LINKED ISSUE #14676: Extract S3 Lineage of Athena table based on DDL
**Feature**
S3 bucket lineage of Athena table based on DDL

**Describe the task**
Currently the lineage ingestion pipeline parses Athena queries to get the table related sources and destinations. But the S3 source can be obtained from the table DDL (`LOCATION` parameter).

### PR DESCRIPTION
**Pull request by harshsoni2024**
### Describe your changes:

Fixes #14676 

I worked on ... because ...

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by sushi30 — CHANGES_REQUESTED** [review]
lets add a test case for this

**Comment by sushi30** [comment]
we can use [localstack athena](https://docs.localstack.cloud/user-guide/aws/athena/) similar to the way [dynamo db is tested](https://github.com/open-metadata/OpenMetadata/blob/7dad49640d326e595c927f0cbd3b44636ffec1bf/ingestion/tests/integration/profiler/test_dynamodb.py).

**Comment by harshsoni2024** [comment]
> we can use [localstack athena](https://docs.localstack.cloud/user-guide/aws/athena/) similar to the way [dynamo db is tested](https://github.com/open-metadata/OpenMetadata/blob/7dad49640d326e595c927f0cbd3b44636ffec1bf/ingestion/tests/integration/profiler/test_dynamodb.py).

@sushi30 , I had checked the working of mocking athena with localstack and testcontainers. I found out that athena db operations comes under pro account/image. [link](https://docs.localstack.cloud/references/coverage/coverage_athena/)

File: ingestion/src/metadata/ingestion/source/database/athena/utils.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    Gets the view definition
+    """
+    full_view_name = f'"{view_name}"' if not schema else f'"{schema}"."{view_name}"'
+    res = connection.execute(f"SHOW CREATE VIEW {full_view_name}").fetchall()
+    if res:
+        return "\n".join(i[0] for i in res)
+    return None
+
+
+# @reflection.cache
```
Comment (sushi30) [inline ingestion/src/metadata/ingestion/source/database/athena/utils.py:180]: why is there a comment here?

File: ingestion/tests/unit/topology/database/test_athena.py [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+                "http://localhost:8585/api/v1/tables/2c040cf8-432d-4597-9517-4794d6142da3",
+            )
+        ),
+        tableType=TableType.Regular,
+        columns=[
+            Column(
+                name=ColumnName("CUSTOMERID"),
+                displayName=None,
+                dataType=DataType.INT,
+                arrayDataType=None,
```
Comment (sushi30) [inline ingestion/tests/unit/topology/database/test_athena.py:227]: I suggest removing all the `None` fields to make the file more concise. This also allows changing the models without needing to change the test files.

File: ingestion/src/metadata/ingestion/source/database/athena/utils.py (hunk 1 above)
Comment (harshsoni2024) [reply ingestion/src/metadata/ingestion/source/database/athena/utils.py]: I think it was on wrong place.

**Review by sushi30 — APPROVED** [review]

**Pull request merged**